### ISSUE TITLE
[Atom] Data upload vs queue synchronization issue

### ISSUE THREAD

**Issue by jhmueller-huawei** [o3de/o3de#18111]
**Describe the bug**
I found an issue with data upload vs. queue synchronization. In short, the general issue is that when data is uploaded via a ResourcePoolResolver (Buffer- or Image-) in the RootScope and the first Scope that uses this data is run on another queue there is no synchronization happening. The *PoolResolvers ensure to insert Barriers for within queue synchronization correctly, but the required semaphores for between-queue synchronization cannot be inserted partly because by the time the *PoolResolvers do this, it is not known which queue the first Scope using the data upload is running on. This could be resolved if the uploads would properly be added to the FrameGraph as attachments which could then probably also insert barriers properly, but this is not happening currently.

Where is this actually happening currently? As far as I can tell, the RayTracingAccelerationStructurePass is the only pass running on a different queue inside the pipeline at the moment. If the data upload is not already done when it runs, due to lack of synchronization, the resulting acceleration structures are wrong and since this pass may only run once if no geometry is changed in a scene it will not correct itself in following frames. For me this bug therefore appears sometimes in the ASV -> Features -> RayTracingIntersectionShader sample resulting in reflections simply not being visible or only parts of the geometry are visible. Interestingly, the former happens more frequently on Linux, while the latter is what typically happens on Windows. Maybe due to a difference of the NVIDIA drivers of these platforms. The issue can of course be fixed by letting the pass run on the Graphics queue instead of the Compute queue.

**Assets required**
o3de-atom-sampleviewer

**Steps to reproduce**
1. Run the ASV GameLauncher
2. In the menu go to "Samples" -> "Features" -> "RayTracingIntersectionShader"
3. May have to repeat steps 1-2 if it works, otherwise you see missing geometry in the reflection.

**Expected behavior**

![image](https://github.com/o3de/o3de-extras/assets/101700150/c440a682-78b2-4846-a2e5-6a9d9113d9ce)

**Actual behavior**

![image](https://github.com/o3de/o3de-extras/assets/101700150/d32cf924-f0d1-43ea-9e55-bfa5da47f7aa)

**Screenshots/Video**
See above.

**Found in Branch**
development for both o3de and ASV.

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) repository**
Shouldn't matter, problem has been existing for some time, but here we go:
o3de/o3de: c7fb2ee2d08841ee4a40cb94c87373782a6eaf3e
o3de/o3de-atom-sampleviewer: 388f4e516e05143d03a933e24932a1c365daeb9d

**Desktop/Device (please complete the following information):**
 - Device: PC
 - OS: Arch Linux, latest (rolling release)
 - CPU AMD Ryzen 9 7950X3D
 - GPU NVidia RTX 4090
 - Memory 128 GB

**Additional context**
None

**Comment by moudgils** [comment]
Tag @akioCL to take a look

**Comment by moudgils** [comment]
Given that Root scope does the copies in the copy queue maybe we blindly add a fence/sema add the end of root scope to ensure that it finishes execution before the first pass begins. Thoughts @akioCL?

**Comment by jhmueller-huawei** [comment]
The root scope does the copies in the graphics queue, not the copy queue. Wouldn't make sense to have those (usually comparably small) uploads interfere with the asynchronous copy queue used for streaming in textures and mesh data.

The barriers that are inserted already ensure that any following scopes on the graphics queue are correctly synchronized, that's why only the async compute queue has issues. So you suggest a semaphore from the graphics to the compute queue? How would you make sure the wait on the compute queue is inserted before the first scope?
